core/capabilities/webapi/target/capability.go

```diff
@@ -0,0 +1,168 @@
+	return &Capability{
+		capabilityInfo:        capabilityInfo,
+		config:                config,
+		registry:              registry,
+		connectorHandler:      connectorHandler,
+		registeredWorkflows:   make(map[string]struct{}),
+		registeredWorkflowsMu: sync.Mutex{},
+		lggr:                  lggr,
+	}, nil
+}
+
+func (c *Capability) Start(ctx context.Context) error {
+	return c.registry.Add(ctx, c)
+}
+
+func (c *Capability) Close() error {
+	return nil
+}
+
+func (c *Capability) Info(ctx context.Context) (capabilities.CapabilityInfo, error) {
+	return capabilityInfo, nil
+}
+
+func getMessageID(req capabilities.CapabilityRequest) (string, error) {
+	if err := validation.ValidateWorkflowOrExecutionID(req.Metadata.WorkflowID); err != nil {
+		return "", fmt.Errorf("workflow ID is invalid: %w", err)
+	}
+	if err := validation.ValidateWorkflowOrExecutionID(req.Metadata.WorkflowExecutionID); err != nil {
+		return "", fmt.Errorf("workflow execution ID is invalid: %w", err)
+	}
+	messageID := []string{
+		req.Metadata.WorkflowID,
+		req.Metadata.WorkflowExecutionID,
+		webcapabilities.MethodWebAPITarget,
+	}
+	return strings.Join(messageID, "/"), nil
+}
+
+func (c *Capability) Execute(ctx context.Context, req capabilities.CapabilityRequest) (capabilities.CapabilityResponse, error) {
+	c.lggr.Debugw("executing http target", "capabilityRequest", req)
+
+	var input Input
+	err := req.Inputs.UnwrapTo(&input)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	var workflowCfg WorkflowConfig
+	err = req.Config.UnwrapTo(&workflowCfg)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	messageID, err := getMessageID(req)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	c.registeredWorkflowsMu.Lock()
+	defer c.registeredWorkflowsMu.Unlock()
+	if _, ok := c.registeredWorkflows[req.Metadata.WorkflowID]; !ok {
+		return capabilities.CapabilityResponse{}, fmt.Errorf("workflow is not registered: %v", req.Metadata.WorkflowID)
+	}
+
+	payload := webcapabilities.TargetRequestPayload{
+		URL:       input.URL,
+		Method:    input.Method,
+		Headers:   input.Headers,
+		Body:      []byte(input.Body),
+		TimeoutMs: workflowCfg.TimeoutMs,
+	}
+
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	switch workflowCfg.Schedule {
+	case SingleNode:
+		// blocking call to handle single node request. waits for response from gateway
+		resp, err := c.connectorHandler.HandleSingleNodeRequest(ctx, messageID, payloadBytes)
+		if err != nil {
+			return capabilities.CapabilityResponse{}, err
+		}
+		c.lggr.Debugw("received gateway response", "resp", resp)
+		var payload webcapabilities.TargetResponsePayload
+		err = json.Unmarshal(resp.Body.Payload, &payload)
+		if err != nil {
+			return capabilities.CapabilityResponse{}, err
+		}
+
+		// TODO: check target response format and fields
```

Comment:
When/how will this TODO get resolved?

Reply:
I propose this struct. Thoughts? 

```
type CapabilityResponsePayload struct {
	Success      bool     
	ErrorMessage string
	StatusCode   uint8   
	Headers      map[string]string 
	Body         string
}
```

Reply:
Is this meant to mirror an HTTP Response?  If so, then Success and ErrorMessage don't seem relevant.

Reply:
`Success` indicates high-level status. HTTPTarget may result in failure before it even makes a HTTP request (e.g. validation error), then the overall `Success` will be `false` and `StatusCode`, `Headers`, `Body` will be empty.

Reply:
I will create a ticket for this for now. so far, all targets have been modeled as async operations. There is a larger discussion on changing targets to sync operations (or just removing targets and making them actions) so that "write" capabilities can be chained.

Reply:
https://smartcontract-it.atlassian.net/browse/CM-473